charts/consul/templates/api-gateway-gatewayclassconfig.yaml

```diff
@@ -11,6 +11,11 @@ metadata:
     component: api-gateway
 spec:
   consul:
+    {{- if .Values.externalServers.enabled }}
+    address: {{ first .Values.externalServers.hosts }}
```

Comment:
Actually, thinking this through a bit more, I don't think we should fully reduce what the previous PR had.

Specifically, our last release (0.4) supported agent-based "externalServers" configurations because it still just talked to the local agent which could directly health-check the deployments. This change means that that setup would potentially no longer work because of the lack of network connectivity between Consul and the deployment in a setup where the deployment is in some sort of restricted network segment.

I think what this means is that we should _only_ be leveraging `externalServers` if both `externalServers` is specified and you're running Agentless -- which would still give us the ability to support 0.4 with `externalServers` but still just say, "You want to run agentless? You'll need to upgrade your gateways."

Reply:
You should be able to verify that this behavior would break here by deploying `hashicorp/consul-api-gateway:0.4` with 0.49 and 1.0.0-beta5 (i.e. this change) with node-based agents + external servers configurations.

Reply:
Sounds good! I added the check for clients. I didn't add the checks that you had for `client.enabled` being set to `"-"` because the expected value is a boolean. I figure if the user is setting the value to `"-"`, they are holding the Helm chart wrong. But I am willing to add the check for `"-"` equivalency if you think that is needed.

Reply:
Cool, sounds good. I added a quick comment about simplifying the logic to avoid some of the compound booleans for clarity + comments, but generally this looks like you're capturing the backwards-compat case we were talking about.